src/mococrw/signature.h

```diff
@@ -34,136 +35,36 @@ class SignatureUtils
+     * Verifies a signature based on the message digest and a given verification context, that
+     * includes an asymmetric public key or X509 certificate (RSA/ECC) .
      *
-     * @param publicKey The public key to be used in the verification
-     * @param padding The padding mode
-     * @param signature The signature to be verified
-     * @param message The plain text message the signature is verified with
-     * @throw MoCOCrWException if the verification fails.
-     */
-    static void verify(AsymmetricPublicKey &publicKey,
-                       const RSAPadding &padding,
-                       const std::vector<uint8_t> signature,
-                       const std::string message);
-
-    /**
-     * @brief Verify a signature
-     *
-     * Verifies a signature based on the message digest and a given public key and padding mode.
-     *
-     * @param publicKey The public key to be used in the verification
-     * @param padding The padding mode
+     * @param ctx Context used to verify the signature
      * @param signature The signature to be verified
      * @param messageDigest The message digest the signature is verified with
      * @throw MoCOCrWException if the verification fails.
      */
-    static void verify(AsymmetricPublicKey &publicKey,
-                       const RSAPadding &padding,
-                       const std::vector<uint8_t> signature,
-                       const std::vector<uint8_t> messageDigest);
-
-    /**
-     * @brief Verify a signature
-     *
-     * Verifies a signature based on the signed message and a given X509 certificate and padding
-     * mode.
-     *
-     * @param certificate A x509 certificate from which the public key will be extracted
-     * @param padding The padding mode
-     * @param signature The signature to be verified
-     * @param message The plain text message the signature is verified with
-     * @throw MoCOCrWException if the verification fails.
-     */
-    static void verify(const X509Certificate &certificate,
-                       const RSAPadding &padding,
-                       const std::vector<uint8_t> signature,
-                       const std::string message);
-
-    /**
-     * @brief Verify a signature
-     *
-     * Verifies a signature based on the message digest and a given X509 certificate and padding
-     * mode.
-     *
-     * @param certificate A x509 certificate from which the public key will be extracted
-     * @param padding The padding mode
-     * @param signature The signature to be verified
-     * @param messageDigest The message digest the signature is verified with
-     * @throw MoCOCrWException if the verification fails.
-     */
-    static void verify(const X509Certificate &certificate,
-                       const RSAPadding &padding,
-                       const std::vector<uint8_t> signature,
-                       const std::vector<uint8_t> messageDigest);
-
+    static void verify(AsymmetricVerifyCtx &ctx,
```

Comment:
Why is this here? For Encrypt/Decrypt this is encapsulated in the params class or the context?